src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -445,6 +446,17 @@ static StoragePtr create(const StorageFactory::Arguments & args)
     }
     else if (merging_params.mode == MergeTreeData::MergingParams::Replacing)
     {
+        // if there is args and number of optional parameter is higher than 1
+        // is_deleted is not allowed with the 'allow_deprecated_syntax_for_merge_tree' settings
+        if (arg_cnt - arg_num == 2 && !engine_args[arg_cnt - 1]->as<ASTLiteral>() && !args.getLocalContext()->getSettingsRef().allow_deprecated_syntax_for_merge_tree)
+        {
```

Comment:
I just implement it on the way we agree on [in this previous comment](https://github.com/ClickHouse/ClickHouse/pull/41005#issuecomment-1252154872)
Moreover, I find it more intuitive to add optional parameter in the end of the parameter list instead of at the beginning (or in middle in the case of Replicated).

Reply:
Sorry, I got confused because I did not notice that arguments are parsed in reverse order here